rundeckapp/grails-app/services/rundeck/services/audit/AuditEventsService.groovy

```diff
@@ -0,0 +1,396 @@
+package rundeck.services.audit
+
+import com.dtolabs.rundeck.core.audit.AuditEvent
+import com.dtolabs.rundeck.core.common.Framework
+import com.dtolabs.rundeck.plugins.audit.AuditEventListener
+import grails.gorm.transactions.Transactional
+import org.apache.log4j.Logger
+import org.springframework.context.event.EventListener
+import org.springframework.core.task.AsyncTaskExecutor
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor
+import org.springframework.security.authentication.event.AuthenticationFailureBadCredentialsEvent
+import org.springframework.security.authentication.event.AuthenticationSuccessEvent
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.web.authentication.logout.LogoutHandler
+import rundeck.services.FrameworkService
+
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+import java.util.concurrent.CopyOnWriteArrayList
+import java.util.stream.Collectors
+
+/**
+ *
+ * Service for capturing and firing auditing events like login, logout or resource access
+ *
+ * @author Alberto Hormazabal
+ *
+ */
+@Transactional
+class AuditEventsService
```

Comment:
This class should not be a grails Service, as it does not need any hibernate session/database access.  It can be moved to the src/groovy directory, and declared in the resources.groovy as a normal spring bean.  As a grails Service, all the method calls to this class are proxied and run within a database transaction